api/src/wfl/module/covid.clj

```diff
@@ -652,52 +701,80 @@
   (update-method-configuration! executor reference)
   (firecloud/submit-method workspace methodConfiguration))
 
-(defn ^:private write-workflows!
-  "Write `workflows` to `details` table."
+(defn ^:private allocate-submission
+  "Write or allocate workflow records for `submission` in `details` table."
   [{:keys [details]                :as _executor}
    {:keys [referenceId]            :as _reference}
    {:keys [submissionId workflows] :as _submission}]
-  (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
-    (let [now     (OffsetDateTime/now)
-          to-rows (fn [{:keys [status workflowId] :as _workflow}]
-                    {:snapshot_reference_id referenceId
-                     :rawls_submission_id   submissionId
-                     :workflow_id           workflowId
-                     :workflow_status       status
-                     :updated               now})]
-      (jdbc/insert-multi! tx details (map to-rows workflows)))))
+  (letfn [(to-row [now {:keys [status workflowId] :as _workflow}]
+            {:reference  referenceId
+             :submission submissionId
+             :workflow   workflowId
+             :status     status
+             :updated    now})]
+    (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+      (->> (map (partial to-row (utc-now)) workflows)
+           (jdbc/insert-multi! tx details)))))
+
+;; Workflows in newly created firecloud submissions may not have been scheduled
+;; yet and thus do not have a workflow uuid. Thus, we first "allocate" the
+;; workflow in the database and then later assign workflow uuids.
+(defn ^:private update-unassigned-workflow-uuids!
+  "Assign workflow uuids from previous submissions"
+  [{:keys [workspace details] :as executor}]
+  (letfn [(read-a-submission-without-workflows []
+            (let [query "SELECT id, submission FROM %s WHERE submission IN (
```

Comment:
What's the reason for this nested query rather than `SELECT id, submission from %s WHERE workflow IS NULL LIMIT 1`?